File: testdata/bench/rust/BenchmarkTest00007.rs
use actix_web::{web, HttpResponse};
use std::collections::HashMap;

// Vulnerable: SQL injection via diesel::sql_query with format
async fn handler(query: web::Query<HashMap<String, String>>) -> HttpResponse {
    let param = query.get("id").unwrap_or(&String::new()).clone();
    let sql = format!("DELETE FROM records WHERE id = {}", param);
    // diesel::sql_query(&sql).execute(&conn)
    HttpResponse::Ok().body(sql)
}

File: testdata/bench/javascript/BenchmarkTest00007.js
// Vulnerable: SQL injection via Sequelize raw query
const express = require('express');
const sequelize = require('./sequelize');
const router = express.Router();
router.get('/test007', (req, res) => {
    const param = req.query.username;
    sequelize.query("SELECT * FROM users WHERE username = '" + param + "'")
        .then(results => res.json(results))
        .catch(err => res.status(500).json({ error: 'Query failed' }));
});
module.exports = router;
